core/src/main/scala/stryker4s/extension/TreeExtensions.scala

```diff
@@ -28,25 +29,36 @@ object TreeExtensions {
         * @return A Some of the parent if the given term is a partial statement,
         *         else a None if the given term is a full statement
         */
-      def unapply(term: Term): Option[Term] = term.parent collect {
+      final def unapply(term: Term): Option[Term] = term.parent collect {
         case parent: Term.Apply      => parent
         case parent: Term.Select     => parent
         case parent: Term.ApplyType  => parent
         case parent: Term.ApplyInfix => parent
       }
     }
 
-    /** Extractor object to check if the [[scala.meta.Term]] is a literal inside a pattern match
+    /** Extractor object to check if the [[scala.meta.Term]] is inside a pattern match
       *
       */
-    private object LiteralPatternMatch {
-
-      def unapply(literal: Lit): Option[Term] = literal.parent match {
-        case Some(parent: Case) =>
-          parent.parent collect {
-            case topParent: Term => topParent.topStatement()
-          }
-        case _ => None
+    private object ParentIsPatternMatch {
+
+      final def unapply(term: Term): Option[Term] = if (term.isIn[Case]) parentOf(term) else None
+
+      /** Go up the tree, until a Case is found, then go up until a `Term` is found
+        *
+        */
+      @tailrec
+      private def parentOf(tree: Tree): Option[Term] = tree.parent match {
+        case Some(caseTree: Case) => findInsideCase(caseTree) // Case is found, go into that
+        case Some(other)          => parentOf(other) // Keep going up
+        case None                 => None // Top of tree is reached, we better stop
+      }
+
+      @tailrec
+      private def findInsideCase(caseTree: Tree): Option[Term] = caseTree.parent match {
+        case Some(term: Term) => Some(term) // Great success!
+        case Some(other)      => findInsideCase(other) // Keep going up
```

Comment:
`topStatement()` doesn't have comments explaining what each case does. I like keeping comments for clarity though, but consistency might be a thing? 
Also, `// Great success!` might not be very descriptive. ^^

Reply:
Fixed. And much more...